Whiskey/Functions/Find-WhiskeyPowerShellModule.ps1

```diff
@@ -41,46 +41,119 @@ function Find-WhiskeyPowerShellModule
+
+            [Parameter(Mandatory)]
+            [Version] $MinimumVersion,
 
-        if( -not [Management.Automation.WildcardPattern]::ContainsWildcardCharacters($version) -and [Version]::TryParse($Version,[ref]$null) )
+            [Parameter(Mandatory)]
+            [Version] $MaximumVersion
+        )
+
+        Write-WhiskeyDebug "Importing module $($Name) $($MinimumVersion) <= $($MaximumVersion)."
+        $module = Get-Module -Name $Name
+        if( $module )
         {
-            $tempVersion = [Version]$Version
-            if( $TempVersion -and ($TempVersion.Build -lt 0) )
+            if( $module.Version -ge $MinimumVersion -and $module.Version -le $MaximumVersion )
             {
-                $Version = [Version]('{0}.{1}.0' -f $TempVersion.Major, $TempVersion.Minor)
+                Write-WhiskeyDebug "Module $($Name) $($module.Version) already imported."
+                return
             }
+
+            Write-WhiskeyDebug "Removing module $($Name) $($module.Version)."
+            $module | Remove-Module -Force
         }
+        
+        $importFailed = $true
+        try
+        {
+            Import-Module -Name $Name `
+                          -MinimumVersion $MinimumVersion `
+                          -MaximumVersion $MaximumVersion `
+                          -Global
+            $importFailed = $false
+        }
+        finally
+        {
+            if( $importFailed )
+            {
+                Write-WhiskeyDebug "Exception importing $($Name) module: $($Global:Error[0])"
 
-        $module = 
-            Find-Module -Name $Name -AllVersions -AllowPrerelease:$AllowPrerelease |
-            Where-Object { $_.Version.ToString() -like $Version } |
-            Sort-Object -Property 'Version' -Descending
+                Write-WhiskeyDebug '\PSModulePath\' -Indent
+                $env:PSModulePath -split ([IO.Path]::PathSeparator) | Write-WhiskeyDebug
+                Write-WhiskeyDebug '/PSModulePath/' -Outdent
+
+                Write-WhiskeyDebug '\Get-Module\' -Indent
+                Get-Module | Out-String | Write-WhiskeyDebug
+                Write-WhiskeyDebug '/Get-Module/' -Outdent
+
+                Write-WhiskeyDebug '\Get-Module -ListAvailable\' -Indent
+                Get-Module -ListAvailable | Out-String | Write-WhiskeyDebug
+                Write-WhiskeyDebug '/Get-Module -ListAvailable/' -Outdent
+            }
+        }
     }
-    else
+
+    try
     {
-        $atVersionString = ''
-        $module = Find-Module -Name $Name -AllowPrerelease:$AllowPrerelease -ErrorAction Ignore
-    }
+        Register-WhiskeyPSModulePath -PSModulesRoot $BuildRoot
+
+        # If you want to upgrade the PackageManagement and PowerShellGet versions, you must also update:
+        # * Install-WhiskeyRequiredModule
+        Import -Name 'PackageManagement' -MinimumVersion '1.3.2' -MaximumVersion '1.4.7'
+        Import -Name 'PowerShellGet' -MinimumVersion '2.0.0' -MaximumVersion '2.2.5'

```

Comment:
You can also just put this into a function that returns the necessary versions of these modules instead of having to leave comments to ensure they match everywhere.

Reply:
I'm going to move them to module-level variables. These versions used to be defined in YAML files and outside the Whiskey module. They aren't anymore.